test/controllers/register/trustees/individual/TrusteeNameControllerSpec.scala

```diff
@@ -0,0 +1,155 @@
+  import TrusteeNameControllerSpec._
+
+  def controller(dataRetrievalAction: DataRetrievalAction = getMandatoryEstablisherCompany): TrusteeNameController =
+    new TrusteeNameController(
+      frontendAppConfig,
+      messagesApi,
+      mockUserAnswersService,
+      new FakeNavigator(desiredRoute = onwardRoute),
+      FakeAuthAction,
+      dataRetrievalAction,
+      FakeAllowAccessProvider(),
+      new DataRequiredActionImpl,
+      formProvider
+    )
+
+  private val postCall = routes.TrusteeNameController.onSubmit _
+
+  def viewAsString(form: Form[_] = form): String = trusteeName(
+    frontendAppConfig,
+    form,
+    NormalMode,
+    firstTrusteeIndex,
+    None,
+    postCall(NormalMode, firstTrusteeIndex, None),
+    None
+  )(fakeRequest, messages).toString
+
+  private val postRequest = fakeRequest.withFormUrlEncodedBody(("firstName", "testFirstName"), ("lastName", "testLastName"))
+
+  "TrusteeNameController" must {
+    "return OK and the correct view for a GET" in {
+      val result = controller().onPageLoad(NormalMode, firstTrusteeIndex, None)(fakeRequest)
+
+      status(result) mustBe OK
+      contentAsString(result) mustBe viewAsString()
+    }
+
+    "populate the view correctly on a GET when the question has previously been answered" in {
+      val validData = Json.obj(
+        TrusteesId.toString -> Json.arr(
+          Json.obj("trustee" -> Json.obj(
+            TrusteeNameId.toString -> personName
+          ))
+        )
+      )
+
+      val getRelevantData = new FakeDataRetrievalAction(Some(validData))
+
+      val result = controller(getRelevantData).onPageLoad(NormalMode, firstTrusteeIndex, None)(fakeRequest)
+
+      contentAsString(result) mustBe viewAsString(form.fill(PersonName("John", "Doe")))
+    }
+
+    "redirect to the next page when valid data is submitted" in {
+      val validData = Json.obj(
+        TrusteesId.toString -> Json.arr(
+          Json.obj("trustee" -> Json.obj(
+            TrusteeNameId.toString -> personName
+          ))
+        )
+      )
+
+      when(mockUserAnswersService.upsert(any(), any(), any())(any(), any(), any())).thenReturn(Future.successful(validData))
+
+      val result = controller().onSubmit(NormalMode, firstTrusteeIndex, None)(postRequest)
+
+      status(result) mustBe SEE_OTHER
+    }
+
+    "return a Bad Request and errors when invalid data is submitted" in {
+      val postRequest = fakeRequest.withFormUrlEncodedBody(("firstName", "01"), ("lastName", "?&^%$£"))
+
+      val result = controller().onSubmit(NormalMode, firstTrusteeIndex, None)(postRequest)
+
+      status(result) mustBe BAD_REQUEST
+    }
+
+    "return a Bad Request and errors when no data is submitted" in {
+      val result = controller().onSubmit(NormalMode, firstTrusteeIndex, None)(fakeRequest)
+
+      status(result) mustBe BAD_REQUEST
+    }
+
+    "save the IsTrusteeNewId flag and when the new trustee is being added" in {
+      reset(mockSectionComplete, mockUserAnswersService)
```

Comment:
Great practice even though the mocks are the same! 🙌 
There is also `BeforeAndAfterEach` from ScalaTest `import org.scalatest.{BeforeAndAfter, BeforeAndAfterEach}` that allows us to register a function to reset the mock.

Reply:
👍🏼